include/iio/iio.h

```diff
@@ -1142,7 +1151,8 @@ iio_buffer_get_channels_mask(const struct iio_buffer *buf);
  * @return On success, a pointer to an iio_block structure
  * @return On failure, a pointer-encoded error is returned */
 __api __check_ret struct iio_block *
-iio_buffer_create_block(struct iio_buffer *buffer, size_t size);
+iio_buffer_create_block(struct iio_buffer *buffer,
+          struct iio_block_params *params);
```

Comment:
While it made sense to group `continuous_dma_memory_required `with `idx `(belonging to iio buffer) it is not the same for the `size `(belonging to the iio block). `size` is a mandatory parameter to be used each time a block is created which is not the case for `continuous_dma_memory_required`.
Please add a new parameter to the `iio_buffer_create_block()` and `iio_buffer_create_stream()` that is an enumeration. The first item in the enumeration should be the default system allocator and the 2nd should be the CMA allocator.
This way we can extend to more allocators if we need to.